src/d3a/models/market/two_sided.py

```diff
@@ -252,10 +252,10 @@ def match_recommendation(self, recommended_list):
                 propagated_bid_rate=bid.price/bid.energy,
                 original_offer_rate=offer.original_offer_price/offer.energy,
                 propagated_offer_rate=offer.price/offer.energy,
-                trade_rate=recommended_pair.trade_rate)
+                trade_rate=original_bid_rate)
 
             bid_trade, trade = self.accept_bid_offer_pair(
-                bid, offer, trade_bid_info.trade_rate, trade_bid_info, selected_energy
+                bid, offer, bid.energy_rate, trade_bid_info, selected_energy
```

Comment:
The same applies here of course. But probably I am missing something here. Please fill me in. Thanks a lot!

Reply:
Well, this is far more straightforward than the former comment. In our markets, we always clear based on the bid rate and not the offer rate. Therefore the trade rate should be the bid rate. However, the myco should have already handled that, therefore I think it is much better to use the `recommended_pair.trade_rate` instead. Will apply the change now, thanks!

Reply:
OK, I am a bid confused by your comment `In our markets, we always clear based on the bid rate and not the offer rate`. What about the pay-as-clear market? Isn't the clearing rate applied However, `recommended_pair` is set correctly for this in the PAC case AFAIS, so your proposed change is adapting for this. Thanks!

Reply:
You are right, my comment only applies for the pay as bid case, for the pay as clear we really need the clearing rate. Nevertheless should be fixed now, thanks!